aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsCookieProcessor.java

```diff
@@ -0,0 +1,267 @@
+
+    // BitSet to validate domain characters
+    static final BitSet domainValid = createDomainValidSet();
+
+    static final DateTimeFormatter COOKIE_DATE_FORMATTER = DateTimeFormatter.RFC_1123_DATE_TIME.withZone(ZoneId.of("GMT"));
+
+    static final String ANCIENT_DATE = COOKIE_DATE_FORMATTER.format(Instant.ofEpochMilli(10000));
+
+    static BitSet createTokenValidSet() {
+        BitSet tokenSet = new BitSet(128);
+        for (char c = '0'; c <= '9'; c++) tokenSet.set(c);
+        for (char c = 'a'; c <= 'z'; c++) tokenSet.set(c);
+        for (char c = 'A'; c <= 'Z'; c++) tokenSet.set(c);
+        for (char c : "!#$%&'*+-.^_`|~".toCharArray()) tokenSet.set(c);
+        return tokenSet;
+    }
+
+    static BitSet createDomainValidSet() {
+        BitSet domainValid = new BitSet(128);
+        for (char c = '0'; c <= '9'; c++) domainValid.set(c);
+        for (char c = 'a'; c <= 'z'; c++) domainValid.set(c);
+        for (char c = 'A'; c <= 'Z'; c++) domainValid.set(c);
+        domainValid.set('.');
+        domainValid.set('-');
+        return domainValid;
+    }
+
+    private final Logger log = LoggerFactory.getLogger(AwsCookieProcessor.class);
+
+    @Override
+    public Cookie[] parseCookieHeader(String cookieHeader) {
+        // Return an empty array if the input is null or empty after trimming
+        if (cookieHeader == null || cookieHeader.trim().isEmpty()) {
+            return new Cookie[0];
+        }
+
+        // Parse cookie header and convert to Cookie array
+        return Arrays.stream(cookieHeader.split("\\s*;\\s*"))
+                .map(this::parseCookiePair)
+                .filter(Objects::nonNull) // Filter out invalid pairs
+                .toArray(Cookie[]::new);
+    }
+
+    /**
+     * Parse a single cookie pair (name=value).
+     *
+     * @param cookiePair The cookie pair string.
+     * @return A valid Cookie object or null if the pair is invalid.
+     */
+    private Cookie parseCookiePair(String cookiePair) {
+        String[] kv = cookiePair.split("=", 2);
+
+        if (kv.length != 2) {
+            log.warn("Ignoring invalid cookie: {}", cookiePair);
+            return null;  // Skip malformed cookie pairs
+        }
+
+        String cookieName = kv[0];
+        String cookieValue = kv[1];
+
+        // Validate name and value
+        if (!isToken(cookieName)){
+            log.warn("Ignoring cookie with invalid name: {}={}", cookieName, cookieValue);
+            return null;  // Skip invalid cookie names
+        }
+
+        if (!isValidCookieValue(cookieValue)) {
+            log.warn("Ignoring cookie with invalid value: {}={}", cookieName, cookieValue);
+            return null;  // Skip invalid cookie values
+        }
+
+        // Return a new Cookie object after security processing
+        return new Cookie(SecurityUtils.crlf(cookieName), SecurityUtils.crlf(cookieValue));
+    }
+
+    @Override
+    public String generateHeader(Cookie cookie) {
```

Comment:
nit: This method builds the header string step-by-step but could be refactored for readability.I think we can  extract some repeated logic (like appending attributes) into helper methods to reduce duplication and improve clarity. This will be very great